app/controllers/forumthreads_controller.rb

```diff
@@ -80,6 +85,20 @@ def destroy
     redirect_to @thread.forum
   end
 
+  def search
+  end
+
+  def search_redirect
```

Comment:
I could also do `params.delete_if{|k,v| v==""}` in the controller or at the beginning of the `filter` method once instead of doing the method you suggested multiple times within the model.